.github/workflows/nightly.yml

```diff
@@ -0,0 +1,61 @@
+name: Publish Nightly Builds
+
+on:
+  push:
+    branches:
+      - nightly_build
+
+jobs:
+  build-test-publish:
+    strategy:
+      matrix:
+        include:
+          - runson: macos-latest # mac
+            archive: async-profiler-3.0-macos-arm64.zip
+
+          - runson: ubuntu-latest # linux x64
+            image: docker://public.ecr.aws/async-profiler/asprof-builder-x86:latest
+            archive: async-profiler-3.0-linux-x64.tar.gz
+
+          - runson: codebuild-test-${{ github.run_id }}-${{ github.run_attempt }} # linux arm
```

Comment:
Back to CodeBuild vs. GitHub runners: is there a problem with using CodeBuild both for x86 and ARM? If a build takes longer, I'm curious what exactly takes time.

Reply:
There's no problem, the extra time is from waiting for codebuild to actually pick up and set up the job, which takes about 2 minutes more vs github runners which pick up the job immediately